lib/elixir/lib/kernel/utils.ex

```diff
@@ -22,6 +22,8 @@ defmodule Kernel.Utils do
         _ -> raise ArgumentError, "invalid syntax in defdelegate #{Macro.to_string(fun)}"
       end
 
+    check_for_default_args(args)
```

Comment:
I think we should check the arguments are all variables. It can't be anything else, really. :)